game/src/js/scenes/TitleScene.js

```diff
@@ -41,23 +41,49 @@ export default class TitleScene extends BaseScene {
     });
 
     //controls, credits, offline mode buttons
-    let settingsButton = new Button(this, 450, 350, 'SETTINGS');
-    settingsButton.buttonDown(() => {
-      new DOMModal(this, 'settings', {
-        cancelButtonSelector: '.exit',
-        onCancel: (modal) => {
-          modal.close();
-        }
+    if(ServiceWorker.isSupported()) {
+      const assets = this.cache.json.get('assets');
+      let offlineMode = localStorage.getItem('offlineStatus');
+      offlineMode = (offlineMode == 'true')
+      console.log('heeeeelp' + offlineMode)
+      if (offlineMode == null) {
+        offlineMode = false;
+      }
+      let serviceWorker = new ServiceWorker();
+      let settingsButton = new Button(this, 450, 350, 'SETTINGS', serviceWorker.isRegistered());
```

Comment:
`serviceWorker.isRegistered()` is what should be assigned to `swCheck` in the passed in data to the modal. Buttons don't expect a boolean as 5th parameter.